manager/orchestrator/replicated/services.go

```diff
@@ -157,7 +157,7 @@ func (r *Orchestrator) reconcile(ctx context.Context, service *api.Service) {
 		r.updater.Update(ctx, r.cluster, service, sortedSlots[:specifiedSlots])
 		err = r.store.Batch(func(batch *store.Batch) error {
 			r.deleteTasksMap(ctx, batch, deadSlots)
-			r.deleteTasks(ctx, batch, sortedSlots[specifiedSlots:])
+			r.setTasksDesiredState(ctx, batch, sortedSlots[specifiedSlots:], api.TaskStateRemove)
```

Comment:
Please add a comment.
For bonus points, you can also add comments for the reconcile() function :)